src/Analyser/MutatingScope.php

```diff
@@ -4200,15 +4200,15 @@ private function generalizeVariableTypeHolders(
 
 			$variableTypeHolders[$variableExprString] = new ExpressionTypeHolder(
 				$variableTypeHolder->getExpr(),
-				self::generalizeType($variableTypeHolder->getType(), $otherVariableTypeHolders[$variableExprString]->getType()),
+				self::generalizeType($variableTypeHolder->getType(), $otherVariableTypeHolders[$variableExprString]->getType(), 0),
```

Comment:
I printed the types after generalization in this PR vs. 1.9.x-dev at 9128321b0

most types are identical. the only different are:

```diff
- string(71) "non-empty-array<literal-string&non-falsy-string, mixed>&oversized-array"
+ string(206) "non-empty-array<literal-string&non-falsy-string, non-empty-array<int|string, 0|(non-empty-array<int|string, 0|(non-empty-array<int|string, *ERROR*>&oversized-array)|float>&oversized-array)>>&oversized-array"
```

so after this PR we no longer get MixedType and therefore we get the errors